core/src/main/java/org/elasticsearch/cluster/ClusterStateTaskExecutor.java

```diff
@@ -38,6 +38,13 @@ default boolean runOnlyOnMaster() {
     }
 
     /**
+     * Callback invoked after new cluster state is published. Note that
+     * this method is not invoked if the cluster state was not updated.
+     */
+    default void clusterStatePublished(ClusterState newClusterState) {
```

Comment:
@bleskes I went back and forth, but opted for this simpler interface since for now the use case we have in mind doesn't need the callback on no change.